src/commands/dev/stop.ts

```diff
@@ -0,0 +1,82 @@
+import { CliUx } from '@oclif/core';
+import chalk from 'chalk';
+import inquirer from 'inquirer';
+import { ArchitectError } from '../..';
+import BaseCommand from "../../base-command";
+import { DockerComposeUtils } from "../../common/docker-compose";
+import { RequiresDocker } from '../../common/docker/helper';
+
+export default class DevStop extends BaseCommand {
+  async auth_required(): Promise<boolean> {
+    return false;
+  }
+
+  static description = 'Stop a local deployment';
+  static examples = [
+    'architect dev:stop <local-environment-name>',
+  ];
+
+  static flags = {
+    ...BaseCommand.flags,
+  };
+
+  static args = [{
+    sensitive: false,
+    name: 'name',
+    description: 'Name of local environment',
+    required: false,
+  }];
+
+  async waitForEnviromentToStop(environment: string): Promise<boolean> {
+    let attempts = 0;
+    // Essentially add a 2 mintue timeout
```

Comment:
this is 3 minutes with 1 check per second right?

Reply:
Listen math is hard, but yes it is